hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -132,6 +150,71 @@ public void mint(
 		treasury.setOwnedNfts(treasury.getOwnedNfts() + metadata.size());
 	}
 
+	/**
+	 * Wiping fungible tokens removes the balance of the given account, as well as reducing the total supply.
+	 * @param accountRel - the relationship between the account which owns the tokens and the token
+	 * @param amount - amount to be wiped
+	 */
+	public void wipe(TokenRelationship accountRel, long amount){
+		validateTrue(amount > 0, INVALID_WIPING_AMOUNT, () ->
+				"Cannot wipe " + amount + " units of " + this + " from " + accountRel);
+
+		validateTrue(hasWipeKey(), TOKEN_HAS_NO_WIPE_KEY, () ->
+					"Cannot wipe Tokens without wipe key.");
+
+		validateFalse(treasury.getId().equals(accountRel.getAccount().getId()), CANNOT_WIPE_TOKEN_TREASURY_ACCOUNT, ()->
+				"Cannot wipe treasury account of token.");
+
+		var newTotalSupply = totalSupply - amount;
+		validateTrue( newTotalSupply >= 0, INSUFFICIENT_TOKEN_BALANCE, () ->
+				"Wiping would negate the total supply of the given token.");
+
+		final var newAccBalance  = accountRel.getBalance() - amount;
+		validateTrue(newAccBalance >= 0, INVALID_WIPING_AMOUNT, () ->
+				"Wiping would negate the account balance.");
+
+		accountRel.setBalance(newAccBalance);
+		setTotalSupply(newTotalSupply);
+	}
+
+	/**
+	 * Wiping unique tokens removes the unique token instances, associated to the given account, as well as reducing the total supply.
+	 * @param ownershipTracker - a tracker of changes made to the ownership of the tokens
+	 * @param accountRel - the relationship between the account, which owns the tokens, and the token
+	 * @param serialNumbers - a list of serial numbers, representing the tokens to be wiped
+	 */
+	public void wipe(OwnershipTracker ownershipTracker, TokenRelationship accountRel, List<Long> serialNumbers) {
+		validateTrue(serialNumbers.size() > 0, FAIL_INVALID, () ->
+				"Cannot wipe " + serialNumbers.size() + " number of Unique Tokens.");
+
+		validateTrue(type == TokenType.NON_FUNGIBLE_UNIQUE, FAIL_INVALID, () ->
+				"Non fungible wipe can be invoked only on Non fungible token type.");
+
+		validateTrue(hasWipeKey(), TOKEN_HAS_NO_WIPE_KEY, () ->
+				"Cannot wipe Tokens without wipe key.");
+
+		validateFalse(treasury.getId().equals(accountRel.getAccount().getId()), CANNOT_WIPE_TOKEN_TREASURY_ACCOUNT, ()->
+				"Cannot wipe treasury account of token.");
```

Comment:
Would it be sensible to add `baseWipeValidation` method and place the common checks for fungible/non-fungible wipes?